Router Bug Fix

zhen0 opened the pull request:
PR Checklist:

- [x] add a short description of what's changed to the top of the `CHANGELOG.md`
- [x] add/update tests (or don't, for reasons explained below)

## Describe this PR
- Updates the 'finally' section in App.vue to only add a tenant to the route if headed to the dashboard.  This reduces the relative params navigation error that was causing an occasional blank screen when trying to navigate via a URL. 
- There's still a long loading/auth pattern happening that we should handle in a separate PR
- Closes #324

znicholasbrown commented on src/App.vue:
```diff
@@ -240,8 +240,7 @@ export default {
         this.shown = true
-
-        if (!this.$route.params.tenant) {
+        if (this.isCloud && !this.$route.name && !this.$route.params.tenant) {
```
Hm I don't think this is correct; this shouldn't be restricted to Cloud (tenants exist in Server now so we still want the slug), and the Dashboard is a named route, aliased to root:
```js
// router.js
 {
    name: 'dashboard',
    alias: '/',
    path: '/:tenant?',
    component: () =>
      import(
        /* webpackChunkName: "dashboard" */ '@/pages/Dashboard/Dashboard.vue'
      ),
    beforeEnter: multiguard([authNavGuard, tenantNavGuard])
  }

```

znicholasbrown reviewed (changes requested):
If this fixes the issue the error you're seeing with a blank dashboard then we can confirm this is the correct location but not quite the right fix. This won't push the slug for Server users and I'm not sure when the condition of the unnamed route will be met.

zhen0 commented:
I've updated this @znicholasbrown but still not sure about that router.replace.  The relative params navigation error seems to be fixed and I don't see any missing slug errors but let me know if you see any issues with it. (I did see an occasional missing slug error when switching from server - I've updated the sidenav to deal with that). 

I also added a first iteration of a loading message as I saw a blank screen too many times.  We should build on that but hopefully this can provide users with a sense that their page is loading.
